### PR TITLE
Feat/download html

### PR DESCRIPTION
**Pull request by apryka**
https://metakeen.atlassian.net/browse/FEE-239

### CODE REVIEW

**Review by dariuszlacheta — APPROVED** [review]

File: src/modules/dashboards/saga.ts [hunk 1]
```diff
@@ -697,6 +703,16 @@ export function* cloneDashboard({
   }
 }
 
+function* exportDashboardToHtml({
+  payload,
+}: ReturnType<typeof exportDashboardToHtmlAction>) {
+  const { dashboardId } = payload;
+  const client = yield getContext(KEEN_ANALYSIS);
+  const { projectId, masterKey } = client.config;
+  const codeSnippet = createCodeSnippet({ projectId, masterKey, dashboardId });
```
Comment (maciejrybaniec) [inline src/modules/dashboards/saga.ts:712]: Use `yield` here to test later that `codeSnippet` is passed to  `exportToHtml `

File: src/modules/dashboards/utils/createCodeSnippet.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  dashboardId,
+  type,
+}: {
+  projectId: string;
+  masterKey: string;
+  dashboardId: string;
+  type?: Snippet;
+}) => {
+  if (type === 'head') {
+    return `<script crossorigin src="https://cdn.jsdelivr.net/npm/keen-dashboard-builder@latest/dist/viewer.min.js"></script>`;
```
Comment (maciejrybaniec) [inline src/modules/dashboards/utils/createCodeSnippet.ts:15]:
```
https://cdn.jsdelivr.net/npm/@keen.io/dashboard-creator@latest/dist/public-dashboard/main.min.js
```

File: src/modules/dashboards/utils/createCodeSnippet.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  }
+  if (type === 'body') {
+    return `
+    <div id="root"></div>
+    <script type="text/javascript">
+       new KeenPublicDashboard({
+         container: '#root',
+         dashboardId: ${dashboardId},
+         backend: {
+           analyticsApiUrl: 'staging-api.keen.io',
```
Comment (maciejrybaniec) [inline src/modules/dashboards/utils/createCodeSnippet.ts:25]:
```
api.keen.io
```

File: src/modules/dashboards/utils/createCodeSnippet.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+           dashboardsApiUrl: 'https://blob-service.us-west-2.test.aws.keen.io',
```
Comment (maciejrybaniec) [inline src/modules/dashboards/utils/createCodeSnippet.ts:26]: https://blob-service.keen.io

File: src/modules/dashboards/utils/createCodeSnippet.ts [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+  <body>
+    <div id="root"></div>
+  </body>
+  <script type="text/javascript">
+     new KeenPublicDashboard({
+       container: '#root',
+       dashboardId: ${dashboardId},
+       backend: {
+         analyticsApiUrl: 'staging-api.keen.io',
+         dashboardsApiUrl: 'https://blob-service.us-west-2.test.aws.keen.io',
```
Comment (maciejrybaniec) [inline src/modules/dashboards/utils/createCodeSnippet.ts:60]: Ditto

File: src/modules/dashboards/utils/createCodeSnippet.ts [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+      </script>
+    `;
+  }
+  return `
+  <!doctype html>
+  <html lang="en">
+  <head>
+    <meta charset="utf-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=11" />
+    <title><%= htmlWebpackPlugin.options.title %></title>
```
Comment (maciejrybaniec) [inline src/modules/dashboards/utils/createCodeSnippet.ts:42]: The title should be different than WebPack plugin variable. Please, change it to for example `Public Dashboard`.

**Review by maciejrybaniec — CHANGES_REQUESTED** [review]

**Review by maciejrybaniec — APPROVED** [review]

**Pull request merged**